src/colorize.cr

```diff
@@ -123,43 +123,63 @@ class Object
   include Colorize::ObjectExtensions
 end
 
-struct Colorize::Object(T)
-  private FORE_DEFAULT       = "39"
-  private FORE_BLACK         = "30"
-  private FORE_RED           = "31"
-  private FORE_GREEN         = "32"
-  private FORE_YELLOW        = "33"
-  private FORE_BLUE          = "34"
-  private FORE_MAGENTA       = "35"
-  private FORE_CYAN          = "36"
-  private FORE_LIGHT_GRAY    = "37"
-  private FORE_DARK_GRAY     = "90"
-  private FORE_LIGHT_RED     = "91"
-  private FORE_LIGHT_GREEN   = "92"
-  private FORE_LIGHT_YELLOW  = "93"
-  private FORE_LIGHT_BLUE    = "94"
-  private FORE_LIGHT_MAGENTA = "95"
-  private FORE_LIGHT_CYAN    = "96"
-  private FORE_WHITE         = "97"
-
-  private BACK_DEFAULT       = "49"
-  private BACK_BLACK         = "40"
-  private BACK_RED           = "41"
-  private BACK_GREEN         = "42"
-  private BACK_YELLOW        = "43"
-  private BACK_BLUE          = "44"
-  private BACK_MAGENTA       = "45"
-  private BACK_CYAN          = "46"
-  private BACK_LIGHT_GRAY    = "47"
-  private BACK_DARK_GRAY     = "100"
-  private BACK_LIGHT_RED     = "101"
-  private BACK_LIGHT_GREEN   = "102"
-  private BACK_LIGHT_YELLOW  = "103"
-  private BACK_LIGHT_BLUE    = "104"
-  private BACK_LIGHT_MAGENTA = "105"
-  private BACK_LIGHT_CYAN    = "106"
-  private BACK_WHITE         = "107"
+module Colorize
+  alias Color = ColorANSI | Color256 | ColorRGB
+
+  enum ColorANSI
+    Default      = 39
+    Black        = 30
+    Red          = 31
+    Green        = 32
+    Yellow       = 33
+    Blue         = 34
+    Magenta      = 35
+    Cyan         = 36
+    LightGray    = 37
+    DarkGray     = 90
+    LightRed     = 91
+    LightGreen   = 92
+    LightYellow  = 93
+    LightBlue    = 94
+    LightMagenta = 95
+    LightCyan    = 96
+    White        = 97
+
+    def fore
+      to_i.to_s
+    end
+
+    def back
+      (to_i + 10).to_s
```

Comment:
Maybe the `fore` and `back` methods could take an IO to avoid intermediary string allocations?

```crystal
def back(io : IO)
  (to_i + 10).to_s(io)
end

@back.back(io)
```